### PR TITLE
Add release notes 2.10.21

### PR DESCRIPTION
**Pull request by serinamarie**
[Preview](https://github.com/PrefectHQ/prefect/blob/release-2.10.21/RELEASE-NOTES.md#release-21021)

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,87 @@
...
+<img width="904" alt="prompt choose a deployment" src="https://github.com/PrefectHQ/prefect/assets/42048900/bff5369f-9568-41c9-a2b1-b2ecdd6cd8c8">
+
+For implementation details, see the following pull request:
+- https://github.com/PrefectHQ/prefect/pull/10189
+
+### Enhancements
+- Enable workspace dashboard by default — https://github.com/PrefectHQ/prefect/pull/10202
+- Add `SendgridEmail` notification block — https://github.com/PrefectHQ/prefect/pull/10118
+- Raise state change hook errors during creation if not correctly formatted — https://github.com/PrefectHQ/prefect/pull/9692
+- Improve `prefect deploy nonexistent-entrypoint` `ValueError` - https://github.com/PrefectHQ/prefect/pull/10210
```
Comment (serinamarie) [inline RELEASE-NOTES.md:67]:
```suggestion
- Improve `prefect deploy` nonexistent entrypoint `ValueError` - https://github.com/PrefectHQ/prefect/pull/10210
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,90 @@
 # Prefect Release Notes
 
+## Release 2.10.21
+
+### Deploy deployments prefixed by flow name during `prefect deploy` 
+Users can now specify the deployment to be executed by prefixing the deployment name with the flow name, providing more flexibility and control to the users.
```
Comment (billpalombi) [inline RELEASE-NOTES.md:6]:
```suggestion
You can now specify the deployment to be executed by prefixing the deployment name with the flow name.
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,90 @@
...
+
+To deploy a deployment with the name `my-deployment` for a flow with the name `my-flow`, run the following command:
```
Comment (billpalombi) [inline RELEASE-NOTES.md:8]:
```suggestion
For example, the following command creates a deployment with the name `my-deployment` for a flow with the name `my-flow`:
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,90 @@
...
+```
+
+This is useful when users have multiple deployments named `my-deployment` for different flows.
+
+For implementation details, see the following pull request:
+- https://github.com/PrefectHQ/prefect/pull/10189
+
+## Use environment variables in deployment steps
+
+Prefect now supports the usage of environment variables in deployment steps, allowing users to access environment variables during the `pull` action at runtime or during the `build` and `push` actions when running `prefect deploy`. Particularly useful for CI/CD builds, this provides more versatility in handling different deployment scenarios.
```
Comment (billpalombi) [inline RELEASE-NOTES.md:20]:
```suggestion
Prefect now supports the usage of environment variables in deployment steps, allowing you to access environment variables during the `pull` action at runtime or during the `build` and `push` actions when running `prefect deploy`. Particularly useful for CI/CD builds, this provides more versatility in handling different deployment scenarios. 
```

**Review by billpalombi — COMMENTED** [review]
First pass

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,90 @@
...
+
+### Deploy deployments prefixed by flow name during `prefect deploy` 
+You can now specify the deployment to be executed by prefixing the deployment name with the flow name.
+
+For example, the following command creates a deployment with the name `my-deployment` for a flow with the name `my-flow`:
+```bash
+prefect deploy --name my-flow/my-deployment
+```
+
+This is especially useful when you have many flows with deployments that have the same name.
```
Comment (billpalombi) [inline RELEASE-NOTES.md:13]:
```suggestion
This is especially useful when you have several flows with deployments that have the same name.
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,90 @@
...
+- prefect.deployments.steps.run_shell_script:
+    script: echo "test-'{{ $PREFECT_API_URL }}'"
+    stream_output: true
+```
+
+For implementation details, see the following pull request:
+- https://github.com/PrefectHQ/prefect/pull/10199
+
+### Use `prefect deploy` with multiple deployments with the same name
+Now, when there are multiple deployments with the same name, the `prefect deploy` command prompts you to choose which one to deploy:
```
Comment (billpalombi) [inline RELEASE-NOTES.md:45]:
```suggestion
When there are multiple deployments with the same name, the `prefect deploy` command now prompts you to choose which one to deploy:
```

**Review by billpalombi — COMMENTED** [review]
Second pass

**Review by billpalombi — APPROVED** [review]
lgtm

**Pull request merged**